src/main/java/tracko/logic/commands/order/EditOrderCommand.java

```diff
@@ -181,21 +211,63 @@ public Optional<Address> getAddress() {
             return Optional.ofNullable(address);
         }
 
-        /**
-         * Sets {@code tags} to this object's {@code tags}.
-         * A defensive copy of {@code tags} is used internally.
-         */
-        public void setTags(Set<Tag> tags) {
-            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        public Optional<List<ItemQuantityPair>> getItemList() {
+            return Optional.ofNullable(itemList);
+        }
+
+        public void setItemList(List<ItemQuantityPair> itemList) {
+            this.itemList = itemList;
         }
 
         /**
-         * Returns an unmodifiable tag set, which throws {@code UnsupportedOperationException}
-         * if modification is attempted.
-         * Returns {@code Optional#empty()} if {@code tags} is null.
+         * Updates the item list. This method checks whether the item that is going to be updated is
+         * in the customer's ordered items list; if it is, then it will check whether the values are different
+         * from the original or 0. If the value is zero, the item will be removed from the order list, else
+         * its quantity will be updated. If the item is not in the item list, then the item will be added to
+         * the customer's ordered items list.
+         *
+         * @param orderToEdit The order whose ordered items list is going to be edited.
+         * @param itemToEdit The item that is going to be added, removed or have its quantity edited.
          */
-        public Optional<Set<Tag>> getTags() {
-            return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
+        public void updateItemList(Order orderToEdit, ItemQuantityPair itemToEdit) throws CommandException {
+            List<ItemQuantityPair> orderedItems = orderToEdit.getItemList();
+            boolean hasItemBeenUpdated = false;
+            for (int i = 0; i < orderedItems.size(); i++) {
+                ItemQuantityPair itemInList = orderedItems.get(i);
+
+                // Check whether the item is in the customer's order list.
+                boolean isItemInList = itemInList.getItemName().equalsIgnoreCase(itemToEdit.getItemName());
```

Comment:
For better code quality you can maybe do if 
```
if (!isItemInList) {
    continue
}
```
right after you declare isItemInList here, because if the item is not in the order list, the rest of the operations would not be necessary.

This will cut down the need to check `if(isItemInList...` and `else if (isItemInList...)` for both conditional checks, and make the code a bit cleaner and easier to read. Would also probably reduce the arrow-like code!

Reply:
it completely slipped my mind that `continue` exists in for loops bc i barely ever use it... but thanks, this is super helpful and makes the loop more efficient!